atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraNamespaceDeleter.java

```diff
@@ -0,0 +1,84 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.namespacedeleter.NamespaceDeleter;
+import com.palantir.common.base.Throwables;
+import java.io.IOException;
+import java.util.function.Supplier;
+import org.apache.cassandra.thrift.Compression;
+import org.apache.cassandra.thrift.NotFoundException;
+import org.apache.thrift.TException;
+
+public final class CassandraNamespaceDeleter implements NamespaceDeleter {
+    private final CassandraKeyValueServiceConfig config;
+    private final Supplier<CassandraClient> cassandraClientSupplier;
+    private final Namespace keyspace;
+
+    public CassandraNamespaceDeleter(
+            CassandraKeyValueServiceConfig config, Supplier<CassandraClient> cassandraClientSupplier) {
+        this.config = config;
+        this.cassandraClientSupplier = cassandraClientSupplier;
+        // Namespace performs some handy validation that also ensures we don't allow sql injection characters in the
+        // keyspace!
+        keyspace = Namespace.create(config.getKeyspaceOrThrow());
+    }
+
+    @Override
+    public void deleteAllDataFromNamespace() {
+        try (CassandraClient client = cassandraClientSupplier.get()) {
+            CassandraKeyValueServices.runWithWaitingForSchemas(
+                    () -> dropKeyspace(keyspace, client), config, client, "Dropping keyspace " + keyspace);
+        } catch (Exception e) {
+            throw Throwables.unwrapAndThrowAtlasDbDependencyException(e);
+        }
+    }
+
+    @Override
+    public boolean isNamespaceDeletedSuccessfully() {
+        try (CassandraClient client = cassandraClientSupplier.get()) {
+            client.describe_keyspace(keyspace.getName());
+            return false;
+        } catch (NotFoundException e) {
+            return true;
+        } catch (TException e) {
+            throw Throwables.throwUncheckedException(e);
+        }
+    }
+
+    private static void dropKeyspace(Namespace namespace, CassandraClient client) throws TException {
+        CqlQuery query = CqlQuery.builder()
+                .safeQueryFormat(SchemaBuilder.dropKeyspace(wrapInQuotes(namespace))
+                        .ifExists()
+                        .buildInternal())
+                .build();
+
+        client.execute_cql3_query(query, Compression.NONE, CassandraKeyValueServiceImpl.WRITE_CONSISTENCY);
```

Comment:
why is `WRITE_CONSISTENCY` sufficient here?

Reply:
I had a note to mention this on the PR, and forgot. I discussed with Sam on Cassandra consistency required for dropping keyspace, and he mentioned that I should be able to do a drop on any consistency level. I chose write instead of all since all is likely overkill. Arguably I could also do just ONE if any consistency level is fine, but I arbitrarily decided WRITE was safer.

Reply:
Let's leave some context in the form of a comment, then!